### PR TITLE
It sort of works!!!

### PR DESCRIPTION
**Pull request by designatednerd**

### CODE REVIEW

**Comment by designatednerd** [comment]
@vokal/webandsystems Super low priority but who wants to take a look at some spectacularly terrible FE code?

**Comment by jrit** [comment]
Can I suggest you update the PR with what this fixed and what is still broken?

**Comment by designatednerd** [comment]
Working:
- CSV is parsing correctly
- Graph actually displays bars for the monthly pass analysis
- I think I figured out how javascript functions work
- CSS is marginally less shitty

Pretty much everything else is TK in future pull requests - I've not submitted any of the work I've done on trying to get the overlay to work because...well, it's not working. :smile:

File: ext/js/data-display.js [hunk 1]
```diff
@@ -108,18 +123,134 @@ function DataDisplay(loads, rides) {
     return { bus: busRides, train: trainRides, unknown: unknownRides }
```
Comment (jrit) [inline ext/js/data-display.js]: missing a `;` at eol

File: ext/js/data-display.js [hunk 2]
```diff
@@ -108,18 +123,134 @@ function DataDisplay(loads, rides) {
...
+        case 10:
+          this.monthNamesInPlay.push("Nov");
+          break;
+        case 11:
+          this.monthNamesInPlay.push("Dec");
+          break;
+        default:
+          this.monthNamesInPlay.push("Smarch");
+          break;  
+      }
```
Comment (jrit) [inline ext/js/data-display.js]: You can make this switch substantially more efficient by just using an array to get the months by referencing the position in the array

Comment (designatednerd) [inline ext/js/data-display.js]: Yeah I suppose I should just have it blow up if it hits the default case since I think I've got this working now.

Comment (jrit) [inline ext/js/data-display.js]: and this isn't C, if you try to get an out of bounds index your machine doesn't shut down

**Comment by designatednerd** [comment]
@jrit: Fix pushed.

**Comment by jrit** [comment]
missing a `;`

**Comment by designatednerd** [comment]
Man, not having an IDE to yell at you about how you fucked things up is pretty crappy. Fix pushed.

**Comment by jrit** [comment]
There are linters and stuff that make up for that, but I know what you mean.

**Comment by thoughtpalette** [comment]
F5 + bang head on keyboard, rinse and repeat. c'est la vie

**Comment by scottferg** [comment]
Vi yells at me when something is fucked up.

**Comment by foresmac** [comment]
http://bit.ly/1BPtHcd

**Comment by adambain-vokal** [comment]
I'd be interested in seeing your vim dotfiles someday, if you wouldnt mind
sharing

*-------- *

_Adam Bain_

Systems Engineer

adam.bain@vokal.io

o: 312.582.4643

c: 312.343.7348

vokal.io

**Comment by designatednerd** [comment]
I'm using coda which is great for syntax highlighting, but it doesn't tell me about shit like "Ain't no semicolon there, dumbass".

**Comment by adambain-vokal** [comment]
Sorry directed @scottferg

*-------- *

_Adam Bain_

Systems Engineer

adam.bain@vokal.io

o: 312.582.4643

c: 312.343.7348

vokal.io

**Comment by scottferg** [comment]
It's mostly a matter of setting up linters like @jrit mentioned. I have
some other whizbang features I've added over the years but @wmbest probably
has the best configured Vi.

**Pull request merged**